superset/views/dashboard/api.py

```diff
@@ -320,6 +326,88 @@ def post(self):
         except SQLAlchemyError as e:
             return self.response_422(message=str(e))
 
+    @expose("/", methods=["DELETE"])
+    @protect()
+    @safe
+    @rison(get_delete_ids_schema)
+    def multiple_delete(self, **kwargs):  # pylint: disable=arguments-differ
+        """Delete multiple Dashboards
+        ---
+        delete:
+          parameters:
+          - in: query
+            name: q
+            content:
+              application/json:
+                schema:
+                  type: array
+                  items:
+                    type: integer
+          responses:
+            200:
+              description: Dashboard multiple delete
+              content:
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      message:
+                        type: string
+                      count:
+                        description: Number of deleted dashboards
+                        type: integer
+            401:
+              $ref: '#/components/responses/401'
+            403:
+              description: Dashboard multiple delete
+              content:
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      message:
+                        type: string
+                      count:
+                        description: Number of deleted dashboards
+                        type: integer
+            404:
+              $ref: '#/components/responses/404'
+            422:
+              $ref: '#/components/responses/422'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        query = self.datamodel.session.query(Dashboard).filter(
+            Dashboard.id.in_(kwargs["rison"])
+        )
+        items = self._base_filters.apply_all(query).all()
+        if not items:
+            return self.response_404()
+        delete_count = 0
+        status_code = 200
+        for item in items:
+            try:
+                check_ownership(item)
+                self.datamodel.delete(item, raise_exception=True)
```

Comment:
Instead of deleting as you go along and checking for ownership, it would probably be better to collect the items that are "to be deleted", checking ownership first. After everything checks out, then iterate over the list, or map the list to a collection of ids and do a bulk `delete where id in()`

Reply:
My goal is to make this method generic (charts also have ownership), when using bulk delete on models with relationships we have to explicitly cleanup, on this case `owners` and `slices`. My take is that the performance gain on this kind of operation may not compensate the necessary specific code and complexity increase. Thoughts?

Reply:
Irrespective of performance, I'd still lean on collecting/checking then deleting as opposed to deleting while things are checked. This allows us to perform the delete operation in a single unit of work, as opposed to breaking things up in which case the delete operation will leave things hanging.

Reply:
+1 on checking ownership before deletion. Currently, if the user attempts to delete a mix of dashboards that do and do not own, they will be left in an unclear state. If the user requests to delete something they don't have permission to delete, the request should fail without any dashboards being modified/deleted, IMO.

Reply:
+1 on all or nothing

Reply:
ok, makes sense

Reply:
Updated, it's all or nothing now